### PR TITLE
Switch from chacha20-poly1305-aead to chacha20poly1305 in the default resolver

### PR DESCRIPTION
**Pull request by expenses**
Hi, [`chacha20-poly1305-aead`](https://crates.io/crates/chacha20-poly1305-aead) hasn't been updated for 4 years, so we should switch to [`chacha20poly1305`](https://crates.io/crates/chacha20poly1305) which is actively maintained.

### CODE REVIEW

**Comment by tarcieri** [comment]
Nice, see also this PR to add XChaCha20Poly1305 using the `chacha20poly1305` crate: https://github.com/mcginty/snow/pull/73

File: src/resolvers/default.rs [hunk 1]
```diff
@@ -155,9 +157,13 @@ impl Cipher for CipherChaChaPoly {
...
         let mut buf = Cursor::new(out);
-        let tag = chacha20_poly1305_aead::encrypt(&self.key, &nonce_bytes, authtext, plaintext, &mut buf);
-        let tag = tag.unwrap();
-        buf.write_all(&tag).unwrap();
+        let payload = Payload {
+            msg: plaintext,
+            aad: authtext,
+        };
+        let buffer = ChaCha20Poly1305::new(self.key.into()).encrypt(&nonce_bytes.into(), payload);
+        let buffer = buffer.unwrap();
```
Comment (mcginty) [inline src/resolvers/default.rs]: small aesthetic preference for chaining vs. shadowing, a la

```rust
let buffer = ChaCha20Poly1305::new(self.key.into()).encrypt(&nonce_bytes.into(), payload).unwrap();
```

File: src/resolvers/default.rs [hunk 2]
```diff
@@ -170,15 +176,17 @@ impl Cipher for CipherChaChaPoly {
...
             authtext,
-            &ciphertext[..ciphertext.len()-TAGLEN],
-            &ciphertext[ciphertext.len()-TAGLEN..],
-            &mut buf);
+            &mut buffer,
+            ciphertext[ciphertext.len()-TAGLEN..].into(),
+        );
         match result {
             Ok(_) => {
+                buf.write(&buffer).unwrap();
```
Comment (mcginty) [inline src/resolvers/default.rs]: Can we just do without this `Cursor` now that the new API doesn't demand it? Ideally we can avoid any heap allocations in this method.

Comment (expenses) [reply src/resolvers/default.rs]: We need to pass in a type that implements [`aeda::Buffer`[(https://docs.rs/aead/0.2.0/aead/trait.Buffer.html) and I'm not sure we can without a `Vec`.

Comment (expenses) [reply src/resolvers/default.rs]: No, I've got it actually

**Comment by mcginty** [comment]
Looks so much cleaner than the past implementation we were using, thank you!!

**Comment by expenses** [comment]
@mcginty done.

**Comment by mcginty** [comment]
Merged and edited your commit message to match the PR. Thanks!

**Pull request merged**